vhost-device-console/src/main.rs

```diff
@@ -81,22 +86,64 @@ impl TryFrom<ConsoleArgs> for VuConsoleConfig {
             return Err(Error::SocketCountInvalid(0));
         }
 
-        if (args.backend == BackendType::Nested) && (args.socket_count != 1) {
-            return Err(Error::WrongBackendSocket);
+        if args.backend == BackendType::Nested {
+            if args.socket_count != 1 {
+                return Err(Error::WrongBackendSocket);
+            }
+
+            if (args.tcp_port.as_ref().map_or(false, |s| !s.is_empty()))
+                || (args
+                    .uds_path
+                    .as_ref()
+                    .map_or(false, |path| !path.as_os_str().is_empty()))
+            {
+                return Err(Error::InvalidCmdlineOption);
+            }
+        }
+
+        if args.backend == BackendType::Network
+            && args
+                .uds_path
+                .as_ref()
+                .map_or(false, |path| !path.as_os_str().is_empty())
+        {
+            return Err(Error::InvalidCmdlineOption);
+        }
+
+        if args.backend == BackendType::Uds
+            && args.tcp_port.as_ref().map_or(false, |s| !s.is_empty())
+        {
+            return Err(Error::InvalidCmdlineOption);
```

Comment:
Use `match args.backend`?

Reply:
Thanks for your reply.
I get several high-priority tasks this week, I will make time to implement the code changes as your comments.